13-cqrs/1-basic/1-before/Shipment.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * This is an example of CQS - command query separation
+ *
+ * @author Nadeesha Dilruwan <nadeesha@adsdaq.eu>
+ */
+final class Shipment
+{
+    /**
+     * @var DateTime
+     */
+    private $delivery;
+
+    /**
+     * @var string
+     */
+    private $proofOfDelivery;
+
+    /**
+     * Shipment constructor.
+     * @param string $shipmentID
+     */
+    public function __construct($shipmentID)
+    {
+        // Load current state
+    }
+
+    /**
+     * This method mutates the delivery state
+     *
+     * @param DateTime $delivery
+     * @param string $proofOfDelivery
+     */
+    public function Deliver(DateTime $delivery, $proofOfDelivery)
```

Comment:
why capital D?

Reply:
Why all method names start with a capital letter?

Reply:
ah I'll change all the methods